app/events/views.py

```diff
@@ -12,15 +12,66 @@ def create():
 		location = event_details['location']
 		event_date = event_details['event_date']
 		created_by = g.user
-		event = Events(name=name, description=description, category=category, \
-			location=location, event_date=event_date, created_by=created_by)
-		event.save()
-		res = {
-			"id" : event.id,
-			"name" : event.name,
-			"date_created" : event.date_created,
-			"created_by" : event.created_by.username
+		#check if the user has an event with a similar name and location
+		existing_event = [event for event in g.user.events if event.name == name \
+			and event.location == location]
+		if not existing_event:
+			#create the event if does not exist
+			event = Events(name=name, description=description, category=category, \
+				location=location, event_date=event_date, created_by=created_by)
+			event.save()
+			res = {
+				"id" : event.id,
+				"name" : event.name,
+				"category" : event.category,
+				"created_by" : event.created_by.username,
+				"location" : event.location,
+				"date_created" : event.date_created
+				}
+			return jsonify(res), 201
+		return jsonify({"message" : "you have a similar event in the same location"}), 302
+	return jsonify({"message" : "please login or register  to create an event"}), 401
+
+@events.route('/all')
+def get_all():
+	"""fetch all events available"""
+	events = Events.get_all()
+	if events:
+		event_list = []
+		for event in events:
+			evnt = {
+				"id" : event.id,
+				"name" : event.name,
+				"description" : event.description,
+				"category" : event.category,
+				"location" : event.location,
+				"created_by" : event.created_by.username,
+				"event_date" : event.event_date,
+				"date_created" : event.date_created
 			}
-		return jsonify(res), 201
-	return jsonify({"message" : "please login or register to create an event"})
-	
\ No newline at end of file
+			event_list.append(evnt)
+		return jsonify(event_list), 200
+	return jsonify({"message" : "no events created yet"}), 200
+
+@events.route('/myevents')
```

Comment:
You can try to filter an event by the user_id to get the events of a particular user

Reply:
sure, that is ok too but since i have the `user` object and it has the `events` attribute, i can simply user `user.events` to get that user's  events